gobblin-data-management/src/main/java/gobblin/data/management/conversion/hive/source/HiveSource.java

```diff
@@ -330,8 +331,10 @@ else if (partition.getTPartition().isSetParameters()
   @Override
   public Extractor getExtractor(WorkUnitState state) throws IOException {
     try {
-      return new HiveConvertExtractor(state, getSourceFs());
-    } catch (TException | HiveException e) {
+      return (Extractor) Class.forName(state.getProp(HIVE_SOURCE_EXTRACTOR_TYPE, DEFAULT_HIVE_SOURCE_EXTRACTOR_TYPE))
+          .getConstructor(WorkUnitState.class, FileSystem.class)
```

Comment:
Please do not non-default extractors through reflection as this creates an undocumented api. The recommended approach is to create a factory interface with a create method that takes the necessary parameters.

For example:

  interface HiveConvertExtractorFactory {
     HiveConvertExtractor createExtractor(WorkUnitState state, FileSystem sourceFs)
  } 

Then you can use something like:

  (new ClassAliasResolver<>(HiveConvertExtractorFactory.class)).resolveClass(state.getProp(HIVE_SOURCE_EXTRACTOR_TYPE, DEFAULT_HIVE_SOURCE_EXTRACTOR_TYPE)).newInstance().create(state, getSourceFs())

You can have a look at HttpClientConfiguratorLoader/HttpClientConfigurator for an example.